src/serializer/ResidualFunctions.js

```diff
@@ -169,6 +169,30 @@ export class ResidualFunctions {
     return factoryFunctionInfos;
   }
 
+  // Preserve residual functions' ordering from original source code.
+  // This is necessary to prevent unexpected code locality issues.
+  // [Algorithm] sort function based on following criterias:
+  // 1. source file alphabetically.
+  // 2. start line number.
+  // 3. start column number.
+  _sortFunctionByOriginalOrdering(functionEntries: Array<[BabelNodeBlockStatement, Array<FunctionInstance>]>): void {
+    functionEntries.sort((funcA, funcB) => {
+      const funcALocation = funcA[0].loc;
+      const funcBLocation = funcB[0].loc;
+      if (!funcALocation || !funcBLocation || !funcALocation.source || !funcBLocation.source) {
+        // Preserve the current ordering if there is no source location information available.
+        return -1;
+      }
+      if (funcALocation.source !== funcBLocation.source) {
+        return funcALocation.source.localeCompare(funcBLocation.source);
```

Comment:
Is `localeCompare` the right choice here? I think it uses the current locale.
Of course, comparing file names is tricky. On Windows, it's basically impossible to know how exactly what the NTFS case insensitive comparison does. But on case sensitive systems, I'd think the right thing to do is a basic character-by-character comparison. Not sure how to best express this in JavaScript.

Reply:
Frankly, I do not think the real ordering of the source path matters for this feature. The purpose is grouping all the functions from the same files together during multi-files scenario so that line/column comparison makes sense.
As long as the comparison function is consistent regarding their relative ordering we should be good.

Reply:
Make sense. Except... Introducing this locale-dependency essentially makes Prepack non-deterministic: It may output different files on different machines. I wouldn't promise that Prepack is entirely deterministic otherwise, but we should avoid non-deterministic behavior whenever possible.

Reply:
Agreed. I will fix it in later PR.